samcli/local/docker/container.py

```diff
@@ -180,6 +180,34 @@ def create(self):
 
         return self.id
 
+    def stop(self, time=3):
+        """
+        Stop a container, with a given number of seconds between sending SIGTERM and SIGKILL.
+
+        Parameters
+        ----------
+        time
+            Optional. Number of seconds between SIGTERM and SIGKILL. Effectively, the amount of time
+            the container has to perform shutdown steps. Default: 3
+        """
+        if not self.is_created():
+            LOG.debug("Container was not created, cannot run stop.")
+            return
+
+        try:
+            self.docker_client.containers.get(self.id).stop(timeout=time)
+        except docker.errors.NotFound:
+            # Container is already removed
+            LOG.debug("Container with ID %s does not exist. Cannot stop!", self.id)
+        except docker.errors.APIError as ex:
+            msg = str(ex)
+            removal_in_progress = ("removal of container" in msg) and ("is already in progress" in msg)
+
+            # When removal is already started, Docker API will throw an exception
+            # Skip such exceptions.
+            if not removal_in_progress:
+                raise ex
+
```

Comment:
We should maybe `LOG.debug` something here if removal is in progress

Reply:
Adding this to the `delete` API as well.